### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by trueadm**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,16 +97,63 @@ function Editor() {
...
+where they can be used exclusively. Attempting to use them outside of an update will trigger a runtime error with an appropiate error.
+For those familiar with React Hooks, you can think of these has having a similar functionality (except `$` functions can be used in any order).
+
+### DOM Reconciler
+
+Lexical has its own DOM reconciler that takes a set of Editor States (always the "current" and the "pending") and applies a "diff"
+on them. It then uses this diff to update only the parts of the DOM that need changing. You can think of this as a kind-of virtual DOM,
+except Lexical is able to skip doing much of the diffing work, as it knows what was mutated in a given update. The DOM reconciler
+adopts performance optimizations that benefit the typical heuristics of a content editable – and is able to ensure consistency for
+LTR and RTL langauges automatically.
```
Comment (harish-sethuraman) [inline README.md:136]:
```suggestion
LTR and RTL languages automatically.
```

File: README.md [hunk 2]
```diff
@@ -97,16 +97,63 @@ function Editor() {
...
+
+### Listeners, Node Transforms and Commands
+
+Outside of invoking updates, the bulk of work done with Lexical is via listeners, node transforms and commands. These all stem from
+the editor and are prefixed with `register`. Another important feature is that all the register methods return a function to easily unsubscribe them. For example here is how you listen to an update to a Lexical editor:
+
+```js
+const unregisterListener = editor.registerUpdateListener(({editorState}) => {
+  // An update has occured!
```
Comment (harish-sethuraman) [inline README.md:145]:
```suggestion
  // An update has occurred!
```

File: packages/lexical-website-new/docs/intro.md [hunk 3]
```diff
@@ -4,19 +4,67 @@ sidebar_position: 1
...
+workflow to preventing cascading/waterfalling of updates. You can retrieve the current editor state using `editor.getEditorState()`.
+
+Editor States are also fully serializable to JSON and can easily be serialized back into to editor using `editor.parseEditorState()`.
+
+### Editor Updates
+
+When you want to change something in an Editor State, you must do it via an update, `editor.update(() => {...})`. The closure passed
+to the update call is important. It's a place where you have full "lexical" context of the active editor state, and it exposes
+access to the underling Editor State's node tree. We promote using `$` prefixed functions in this context, as it signifies a place
+where they can be used exclusively. Attempting to use them outside of an update will trigger a runtime error with an appropiate error.
```
Comment (harish-sethuraman) [inline packages/lexical-website-new/docs/intro.md:41]:
```suggestion
where they can be used exclusively. Attempting to use them outside of an update will trigger a runtime error with an appropriate error.
```

File: packages/lexical-website-new/docs/intro.md [hunk 4]
```diff
@@ -4,19 +4,67 @@ sidebar_position: 1
...
+## Lexical is a framework
+
+The core of Lexical is a dependency-free text editor framework that allows developers to build powerful, simple and complex,
+editor surfaces. Lexical's has a few concepts that are worth exploring:
+
+### Editor instances
+
+Editor instances are the core thing that wires everything together. You can attach a contenteditable DOM element to editor instances, and also
+register listeners and commands. Most importantly, the editor allows for updates to its `EditorState`. You can create an editor instance
+using the `createEditor()` API, however you normally don't have to worry when using framworking bindings such as `@lexical/react` as this
```
Comment (harish-sethuraman) [inline packages/lexical-website-new/docs/intro.md:20]:
```suggestion
using the `createEditor()` API, however you normally don't have to worry when using frameworking bindings such as `@lexical/react` as this
```

`framework` might be apt here?

File: README.md [hunk 5]
```diff
@@ -97,16 +97,63 @@ function Editor() {
...
-- a set of editor states that represent the current and pending states of the editor at any given time.
-- a DOM reconciler that takes a set of editor states, diffs the changes, and updates the DOM according to their state.
+### Editor instances
 
-By design, the core of Lexical tries to be as minimal as possible. Lexical doesn't directly concern itself with things that monolithic editors tend to do – such as UI components, toolbars or rich-text features and markdown. Instead the logic for those features can be included via a plugin interface and used as and when they're needed. This ensures great extensibilty and keeps code-sizes to a minimal – ensuring apps only pay the cost for what they actually import.
+Editor instances are the core thing that wires everything together. You can attach a contenteditable DOM element to editor instances, and also
+register listeners and commands. Most importantly, the editor allows for updates to its `EditorState`. You can create an editor instance
+using the `createEditor()` API, however you normally don't have to worry when using framworking bindings such as `@lexical/react` as this
```
Comment (harish-sethuraman) [inline README.md:106]:
```suggestion
using the `createEditor()` API, however you normally don't have to worry when using frameworking bindings such as `@lexical/react` as this
```
`framework` might be apt here as well?

File: README.md [hunk 6]
```diff
@@ -97,16 +97,63 @@ function Editor() {
...
+workflow to preventing cascading/waterfalling of updates. You can retrieve the current editor state using `editor.getEditorState()`.
+
+Editor States are also fully serializable to JSON and can easily be serialized back into to editor using `editor.parseEditorState()`.
+
+### Editor Updates
+
+When you want to change something in an Editor State, you must do it via an update, `editor.update(() => {...})`. The closure passed
+to the update call is important. It's a place where you have full "lexical" context of the active editor state, and it exposes
+access to the underling Editor State's node tree. We promote using `$` prefixed functions in this context, as it signifies a place
+where they can be used exclusively. Attempting to use them outside of an update will trigger a runtime error with an appropiate error.
```
Comment (harish-sethuraman) [inline README.md:127]:
```suggestion
where they can be used exclusively. Attempting to use them outside of an update will trigger a runtime error with an appropriate error.
```

**Review by fantactuka — APPROVED** [review]

File: packages/lexical-website-new/docs/intro.md [hunk 7]
```diff
@@ -4,19 +4,67 @@ sidebar_position: 1
...
+where they can be used exclusively. Attempting to use them outside of an update will trigger a runtime error with an appropiate error.
+For those familiar with React Hooks, you can think of these has having a similar functionality (except `$` functions can be used in any order).
+
+### DOM Reconciler
+
+Lexical has its own DOM reconciler that takes a set of Editor States (always the "current" and the "pending") and applies a "diff"
+on them. It then uses this diff to update only the parts of the DOM that need changing. You can think of this as a kind-of virtual DOM,
+except Lexical is able to skip doing much of the diffing work, as it knows what was mutated in a given update. The DOM reconciler
+adopts performance optimizations that benefit the typical heuristics of a content editable – and is able to ensure consistency for
+LTR and RTL langauges automatically.
```
Comment (harish-sethuraman) [inline packages/lexical-website-new/docs/intro.md:50]:
```suggestion
LTR and RTL languages automatically.
```

File: packages/lexical-website-new/docs/intro.md [hunk 8]
```diff
@@ -4,19 +4,67 @@ sidebar_position: 1
...
+
+Outside of invoking updates, the bulk of work done with Lexical is via listeners, node transforms and commands. These all stem from
+the editor and are prefixed with `register`. Another important feature is that all the register methods return a function to easily unsubscribe them. For example here is how you listen to an update to a Lexical editor:
 
-- editor instances that each attach to a single content editable element.
-- a set of editor states that represent the current and pending states of the editor at any given time.
-- a DOM reconciler that takes a set of editor states, diffs the changes, and updates the DOM according to their state.
+```js
+const unregisterListener = editor.registerUpdateListener(({editorState}) => {
+  // An update has occured!
```
Comment (harish-sethuraman) [inline packages/lexical-website-new/docs/intro.md:59]:
```suggestion
  // An update has occurred!
```

**Comment by trueadm** [comment]
@harish-sethuraman Thanks for all the help! :)

**Comment by harish-sethuraman** [comment]
Sorry for the mess 😅  It doesn't show me option to resolve conversation once it is fixed otherwise I would have closed it 😅

**Pull request merged**